### PR TITLE
Honor probeHTTP code field

### LINKED ISSUE #342: HTTP probe expected code is ignored
The `code` field of `probeHTTP` isn't honoured. I can see from the implementation that this field is passed to the method that executes the probe, but it isn't actually used there, or in the calling function.

Steps to reproduce:
1. Set up an endpoint that returns a 404
2. Configure the following scenario:

```yaml
    scenarios:
    - name: test
      steps:
      # make sure the service responds
      - probeHTTP:
          code: 404
          target:
            service:
              name: my-service
              namespace: my-namespace
              port: 80
```

Expected behaviour:
Probe succeeds

Actual behaviour:
```
owerfulseal-8498cf7884-g2js6 powerfulseal 2021-07-15 18:04:24 INFO action_probe_http.test Making a call: http://<url>, get, {}, 1000, 404, , , True
powerfulseal-8498cf7884-g2js6 powerfulseal 2021-07-15 18:04:24 ERROR action_probe_http.test Exception while calling <url>
powerfulseal-8498cf7884-g2js6 powerfulseal Traceback (most recent call last):
powerfulseal-8498cf7884-g2js6 powerfulseal   File "/usr/local/lib/python3.7/site-packages/powerfulseal/policy/action_probe_http.py", line 94, in make_call
powerfulseal-8498cf7884-g2js6 powerfulseal     resp.raise_for_status()
powerfulseal-8498cf7884-g2js6 powerfulseal   File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 941, in raise_for_status
powerfulseal-8498cf7884-g2js6 powerfulseal     raise HTTPError(http_error_msg, response=self)
powerfulseal-8498cf7884-g2js6 powerfulseal requests.exceptions.HTTPError: 404 Client Error: Not Found for url: <url>
powerfulseal-8498cf7884-g2js6 powerfulseal 2021-07-15 18:04:24 ERROR action_probe_http.Itest No more retries allowed. Failing step
powerfulseal-8498cf7884-g2js6 powerfulseal 2021-07-15 18:04:24 WARNING scenario.test Step returned failure {'probeHTTP': {'code': 404, 'target': {'service': {'name': 'my-service', 'namespace': 'my-namespace', 'port': 80}}}}. Finishing scenario early
```

PR incoming 👍

### PR DESCRIPTION
**Pull request by Niksko**
Closes #342.

This implements handling of the probeHTTP `code` field, which allows the user to specify the expected return code of an HTTP probe. Previously, setting this didn't work, unless the code returned was a non-4xx or 5xx code.

### CODE REVIEW

**Comment by Niksko** [comment]
BTW, thanks for this project. It's got excellent docs, it's simple to configure while still being powerful, and the features set is better aligned with use in big orgs (i.e. not just toys), unlike some of the other popular chaos engineering tools.

File: powerfulseal/policy/action_probe_http.py [hunk 1]
```diff
@@ -78,25 +78,28 @@ def make_call(self, url, method, body, headers, timeout, code, proxy, verify):
...
-                url,
-                headers=headers,
-                timeout=timeout/1000,
-                data=body.encode("utf-8"),
-                proxies=dict(
-                    http=proxy or "",
-                    https=proxy or "",
-                ),
-                verify=verify,
+        resp = requests.request(
```
Comment (jcstanaway) [inline powerfulseal/policy/action_probe_http.py:81]: Should just the call to `requests.request()` be wrapped with a try? Log an error and return False if `requests.request()` itself raises an exception.

Comment (Niksko) [reply powerfulseal/policy/action_probe_http.py:81]: Yeah, probably a good idea. Will do.

**Comment by seeker89** [comment]
> BTW, thanks for this project. It's got excellent docs, it's simple to configure while still being powerful, and the features set is better aligned with use in big orgs (i.e. not just toys), unlike some of the other popular chaos engineering tools.

Thanks a lot, I'm super glad you're liking the docs - that's probably like the biggest lacking point for open source project, and a lot of it was contributed by the great community :)

And yes, it's being used in quite a few big, serious companies, so needs enterprise-ready features :)

**Review by seeker89 — APPROVED** [review]

**Pull request merged**